WordPress/src/main/java/org/wordpress/android/ui/posts/JetpackSecuritySettingsFragment.java

```diff
@@ -0,0 +1,165 @@
+        mJpMonitorWpNotesPref = (WPSwitchPreference) getChangePref(R.string.pref_key_jetpack_send_wp_notifications);
+        mJpSsoPref = (WPSwitchPreference) getChangePref(R.string.pref_key_jetpack_allow_wpcom_sign_in);
+        mJpBruteForcePref = (WPSwitchPreference) getChangePref(R.string.pref_key_jetpack_prevent_brute_force);
+        mJpMatchEmailPref = (WPSwitchPreference) getChangePref(R.string.pref_key_jetpack_match_via_email);
+        mJpUseTwoFactorPref = (WPSwitchPreference) getChangePref(R.string.pref_key_jetpack_require_two_factor);
+    }
+
+    public void setPreferencesFromSiteSettings() {
+        mJpMonitorActivePref.setChecked(mSiteSettings.isJetpackMonitorEnabled());
+        mJpMonitorEmailNotesPref.setChecked(mSiteSettings.shouldSendJetpackMonitorEmailNotifications());
+        mJpMonitorWpNotesPref.setChecked(mSiteSettings.shouldSendJetpackMonitorWpNotifications());
+        mJpBruteForcePref.setChecked(mSiteSettings.isJetpackProtectEnabled());
+        mJpSsoPref.setChecked(mSiteSettings.isJetpackSsoEnabled());
+        mJpMatchEmailPref.setChecked(mSiteSettings.isJetpackSsoMatchEmailEnabled());
+        mJpUseTwoFactorPref.setChecked(mSiteSettings.isJetpackSsoTwoFactorEnabled());
+    }
+
+    private Preference getChangePref(int id) {
+        return WPPrefUtils.getPrefAndSetChangeListener(this, id, this);
+    }
+
+    @Override
+    public boolean onPreferenceChange(Preference preference, Object newValue) {
+        if (newValue == null) {
+            return false;
+        }
+
+        if (preference == mJpMonitorActivePref) {
+            mJpMonitorActivePref.setChecked((Boolean) newValue);
+            mSiteSettings.enableJetpackMonitor((Boolean) newValue);
+        } else if (preference == mJpMonitorEmailNotesPref) {
+            mJpMonitorEmailNotesPref.setChecked((Boolean) newValue);
+            mSiteSettings.enableJetpackMonitorEmailNotifications((Boolean) newValue);
+        } else if (preference == mJpMonitorWpNotesPref) {
+            mJpMonitorWpNotesPref.setChecked((Boolean) newValue);
+            mSiteSettings.enableJetpackMonitorWpNotifications((Boolean) newValue);
+        } else if (preference == mJpBruteForcePref) {
+            mJpBruteForcePref.setChecked((Boolean) newValue);
+            mSiteSettings.enableJetpackProtect((Boolean) newValue);
+        } else if (preference == mJpSsoPref) {
+            mJpSsoPref.setChecked((Boolean) newValue);
+            mSiteSettings.enableJetpackSso((Boolean) newValue);
+        } else if (preference == mJpMatchEmailPref) {
+            mJpMatchEmailPref.setChecked((Boolean) newValue);
+            mSiteSettings.enableJetpackSsoMatchEmail((Boolean) newValue);
+        } else if (preference == mJpUseTwoFactorPref) {
+            mJpUseTwoFactorPref.setChecked((Boolean) newValue);
+            mSiteSettings.enableJetpackSsoTwoFactor((Boolean) newValue);
+        }
```

Comment:
Not a big deal, but I think it would help readability a bit if we extracted out the `Boolean` cast so that we don't have to do the cast repeatedly in the if-else block:

```
Boolean prefBool = (Boolean) newValue;
if (preference == mJpMonitorActivePref) {
    mJpMonitorActivePref.setChecked(prefBool);
    mSiteSettings.enableJetpackMonitor(prefBool);
} else if (preference == mJpMonitorEmailNotesPref) {
    mJpMonitorEmailNotesPref.setChecked(prefBool);
    mSiteSettings.enableJetpackMonitorEmailNotifications(prefBool);
} else if ...
```

If we had written this file in Kotlin, we could have also used a `when` statement to make this a bit more readable:

```
when (preference) {
    mJpMonitorActivePref -> {
        mJpMonitorActivePref.setChecked(prefBool);
        mSiteSettings.enableJetpackMonitor(prefBool);
    }
    mJpMonitorEmailNotesPref -> {
        mJpMonitorEmailNotesPref.setChecked(prefBool);
        mSiteSettings.enableJetpackMonitorEmailNotifications(prefBool);
    }
}
...
```

Reply:
You are right. Thanks! Fixed.